### DISCUSSION TITLE
Add textarea input for config_flow

### DISCUSSION THREAD

**Discussion by alandtse** [home-assistant/frontend#11299] [Other feature requests]
## The request
Config flow should allow integrations to input data from textareas. This may be useful for inputing multiline strings. For example, I am looking to import cookies.txt to allow logging into an integration using cookies saved from the browser. If developers can't specify it via voluptuous, one option may be to default to textarea or to dynamically convert to a textarea instead of a text input if a `\n` is detected.

## The alternatives
We could force users to manually type a `\n` in the string and then reparse that into newlines later. It seems inefficient since text inputs aren't designed for multiline inputs anyway and if it's a large source text blocks it could be incredibly painful for users. Alternatively, we could allow a file upload field and parse the cookies.txt directly.

## Additional information
This allows integrations additional flexibility since they can now effectively parse multi-line strings.

**Comment by KTibow** [comment]
Mind transferring this to a discussion?
